src/modules/users/users.repository.ts

```diff
@@ -26,34 +37,60 @@ export class UsersRepository extends Repository<User> {
   }
 
   async readAllUsers(
-    searchParams: ISearchParams,
+    filterParams: IFilterParams,
     sortParams: ISortParams,
     limit: number,
     skip: number,
   ): Promise<User[]> {
-    const emailMySql = searchParams.email ? `%${searchParams.email}%` : '%';
-    const nameMySql = searchParams.name ? `%${searchParams.name}%` : '%';
-    const surnameMySql = searchParams.surname
-      ? `%${searchParams.surname}%`
+    const emailMySql = filterParams.email ? `%${filterParams.email}%` : '%';
+    const nameMySql = filterParams.name ? `%${filterParams.name}%` : '%';
+    const surnameMySql = filterParams.surname
+      ? `%${filterParams.surname}%`
       : '%';
     const sortByMySql = sortParams.sortBy || 'id';
     const orderByMySql = sortParams.orderBy == 'desc' ? 'DESC' : 'ASC';
     const limitMySql = limit || 1e11;
+    const roleMySql =
+      filterParams.role === RoleEnum.admin ? '%' : RoleEnum.customer;
 
     const users = await this.createQueryBuilder('user')
+      .select([
+        'user.id AS id',
+        'user.name AS name',
+        'user.surname AS surname',
+        'user.email AS email',
+        'user.phone AS phone',
+        'user.password AS password',
+        'role.role_name AS role_name',
+      ])
+      .leftJoin('user.role', 'role')
       .where('user.email LIKE :email', { email: emailMySql })
       .andWhere('user.name LIKE :name', { name: nameMySql })
       .andWhere('user.surname LIKE :surname', { surname: surnameMySql })
+      .andWhere('user.role_id LIKE :role', { role: roleMySql })
```

Comment:
Do not use LIKE here, LIKE is a very slow query inside DB, here is much faster to use `user.role_id = :role`. BTW all field's which is used in WHERE condition should be indexed in DB.

- [ ] double check that this field is indexed